cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceStatusTest.java

```diff
@@ -0,0 +1,200 @@
+
+        JsonObject proposal = buildOptimizationProposal();
+
+        JsonArray loadBeforeArray = proposal.getJsonObject(CruiseControlRebalanceKeys.LOAD_BEFORE_OPTIMIZATION.getKey())
+                .getJsonArray(CruiseControlRebalanceKeys.BROKERS.getKey());
+        JsonArray loadAfterArray = proposal.getJsonObject(CruiseControlRebalanceKeys.LOAD_AFTER_OPTIMIZATION.getKey())
+                .getJsonArray(CruiseControlRebalanceKeys.BROKERS.getKey());
+
+        JsonObject output = KafkaRebalanceAssemblyOperator.parseLoadStats(
+                loadBeforeArray, loadAfterArray);
+
+        assertThat(output.getMap(), hasKey("1"));
+
+        assertThat(output.getJsonObject("1").getMap(), hasKey(CruiseControlLoadParameters.REPLICAS.getKafkaRebalanceStatusKey()));
+
+        JsonObject replicas = output.getJsonObject("1").getJsonObject("replicas");
+
+        ArrayList<Integer> replicasList = new ArrayList<>();
+        replicasList.add(replicas.getInteger("before"));
+        replicasList.add(replicas.getInteger("after"));
+        replicasList.add(replicas.getInteger("diff"));
+
+        assertThat(replicasList.get(0), is(10));
+        assertThat(replicasList.get(1), is(5));
+        assertThat(replicasList.get(2), is(-5));
+
+        assertThat(output.getJsonObject("1").getMap(), hasKey(CruiseControlLoadParameters.CPU_PERCENTAGE.getKafkaRebalanceStatusKey()));
+
+        JsonObject cpus = output.getJsonObject("1").getJsonObject("cpuPercentage");
+
+        ArrayList<Double> cpu = new ArrayList<>();
+
+        cpu.add(cpus.getDouble("before"));
+        cpu.add(cpus.getDouble("after"));
+        cpu.add(cpus.getDouble("diff"));
+
+        assertThat(cpu.get(0), is(10.));
+        assertThat(cpu.get(1), is(20.0));
+        assertThat(cpu.get(2), is(10.0));
+
+    }
+
+    @Test
+    public void testProcessProposal() {
+
+        JsonObject proposal = buildOptimizationProposal();
+
+        KafkaRebalance kr =
+                createKafkaRebalance(CLUSTER_NAMESPACE, CLUSTER_NAME, RESOURCE_NAME, new KafkaRebalanceSpecBuilder().build());
+
+        KafkaRebalanceAssemblyOperator.MapAndStatus<ConfigMap, Map<String, Object>> output = KafkaRebalanceAssemblyOperator.processOptimizationProposal(kr, proposal);
+
+        assertTrue(output.getStatus().containsKey(CruiseControlRebalanceKeys.SUMMARY.getKey()));
+
+        Map<String, String> brokerMap = output.getLoadMap().getData();
+
+        try {
+
+            ObjectMapper mapper = new ObjectMapper();
+
+            Map<String, LinkedHashMap<String, String>> brokerLoadMap = mapper.readValue(brokerMap.get(BROKER_LOAD_KEY), LinkedHashMap.class);
+
+            assertThat(brokerMap, hasKey(BROKER_LOAD_KEY));
+
+            LinkedHashMap<String, LinkedHashMap<String, Object>> m = (LinkedHashMap) brokerLoadMap.get("1");
+
+            assertThat(m, hasKey(CruiseControlLoadParameters.CPU_PERCENTAGE.getKafkaRebalanceStatusKey()));
+
+            ArrayList<Double> cpu = new ArrayList<>();
```

Comment:
Again why not just do: ` assertThat((Double) m.get("cpuPercentage").get("before"), is(10.0));`?